app/grandchallenge/evaluation/forms.py

```diff
@@ -342,19 +349,24 @@ def __init__(self, *args, user, phase: Phase, **kwargs):  # noqa: C901
 
         if self._phase.submission_kind == SubmissionKindChoices.ALGORITHM:
             del self.fields["user_upload"]
-            qs = self.user_active_images_for_phase.order_by("algorithm__title")
+            qs = self.user_algorithms_for_phase.filter(
+                has_active_image=True
+            ).order_by("title")
             if self._phase.parent:
                 qs = qs.filter(
-                    submission__phase=self._phase.parent,
-                    submission__evaluation__status=Evaluation.SUCCESS,
-                    job__status=Job.SUCCESS,
+                    algorithm_container_images__submission__phase=self._phase.parent,
+                    algorithm_container_images__submission__evaluation__status=Evaluation.SUCCESS,
+                    algorithm_container_images__job__status=Job.SUCCESS,
```

Comment:
For the submission to the child phases do we also take into account the algorithm image AND model to the parent phase?

Reply:
Ah good catch. No we don't yet. Just for my own understanding: we want to only allow algorithms that have a successful submission to the parent phase with the currently active image and model (if applicable). That's a more complex query than what we currently have.

Reply:
Indeed - that's correct, see roadmap 301 for the motivation.

Reply:
This is one huge query now, but it's as efficient as it will get if we want to check all this. Something that I noticed just now when checking queries though is that we do two additional lookups per algorithm in order to display the active image pk and active model pk and their comments in the field label. I should annotate those on the queryset instead. I might not get to pushing those changes until Thursday, depends on Loïc